src/System.Management.Automation/engine/Modules/ImportModuleCommand.cs

```diff
@@ -1882,6 +1883,94 @@ internal override IList<PSModuleInfo> ImportModulesUsingWinCompat(IEnumerable<st
         {
             IList<PSModuleInfo> moduleProxyList = new List<PSModuleInfo>();
 #if !UNIX
+            // the ModuleDeny list is cached in PowerShellConfig object
+            string[] moduleDenyList = PowerShellConfig.Instance.GetWindowsPowerShellCompatibilityModuleDenyList();
+
+            // one of the two parameters can be passed: either ModuleNames (most of the time) or ModuleSpecifications (they are used in different parameter sets)
+            List<string> filteredModuleNames = null;
+            if (moduleNames != null)
+            {
+                if ((moduleDenyList == null) || (moduleDenyList.Length == 0))
+                {
+                    filteredModuleNames = new List<string>(moduleNames);
+                }
+                else
+                {
+                    filteredModuleNames = new List<string>();
+                    foreach (var moduleName in moduleNames)
+                    {
+                        // moduleName can be just a module name and it also can be a full path to psd1 from which we need to extract the module name
+                        var exactModuleName = Path.GetFileNameWithoutExtension(moduleName);
```

Comment:
I think we can remove some duplicate code by re-organizing a bit.  We could have a single helper method that determines if a module name matches the deny list, and also emits error as needed:
```
bool IsInDenyList(moduleName);
```
The helper method could then be used to filter each of the separate module name and specification lists.
```
if (!IsInDenyList(moduleName)) { // Add to name or spec filtered list }

Reply:
Added `IsModuleInDenyList` method and templatized parent functionality so that same function is used on both lists.